File: bento/src/main/java/com/yelp/android/bento/componentcontrollers/TabViewPagerComponentController.java
package com.yelp.android.bento.componentcontrollers;

import androidx.annotation.Nullable;
import java.util.ArrayList;
import java.util.List;

public class TabViewPagerComponentController extends ViewPagerComponentController {
    private List<String> mPageTitles = new ArrayList<>();

    public void setPageTitles(List<String> pageTitles) {
        mPageTitles = pageTitles;
    }

    @Nullable
    @Override
    public CharSequence getPageTitle(int position) {
        return mPageTitles.get(position);
    }
}

File: bento/src/main/java/com/yelp/android/bento/componentcontrollers/TabViewPagerComponentController.kt
package com.yelp.android.bento.componentcontrollers

class TabViewPagerComponentController : ViewPagerComponentController() {
    var pageTitles = listOf<String>()

    override fun getPageTitle(position: Int): CharSequence? {
        return pageTitles[position]
    }
}
